Add batch of unary functions

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #64323 Add embedding shape analysis
* #64121 Max Pool with indices
* #63530 Add Maxpool to shape analysis / Opinfo
* #64300 Add symbolic shape comparison optimization
* #64299 Refactor to use shape arguments
* #63516 Add view with negative dim
* #63615 Generalize expand logic
* #63407 Add permute, arange
* #63365 Add support for slice, selec twith int, index_select
* #63099 Add squeeze, unsqueeze, transpose shape functins
* **#63050 Add batch of unary functions**

Differential Revision: [D30347513](https://our.internmc.facebook.com/intern/diff/D30347513)

Commit:
Add batch of unary functions

[ghstack-poisoned]

eellison commented on torch/csrc/jit/runtime/symbolic_shape_registry.cpp:
```diff
@@ -308,11 +309,21 @@ static const OperatorMap<std::string>& get_schema_to_function_graph() {
+      {"aten::erf(Tensor self) -> (Tensor)", "unary"},
+      {"aten::zeros(int[] size, *, int? dtype=None, int? layout=None, Device? device=None, bool? pin_memory=None) -> (Tensor)", "unary_four_unused_inputs"},
+      {"aten::to.dtype(Tensor(a) self, int dtype, bool non_blocking=False, bool copy=False, int? memory_format=None) -> (Tensor(a))", "unary_four_unused_inputs"},
+      {"aten::layer_norm(Tensor input, int[] normalized_shape, Tensor? weight=None, Tensor? bias=None, "
```
![image](https://user-images.githubusercontent.com/11477974/129084474-5e7a6096-4d07-4f70-824c-b9456fecf4d4.png)

![image](https://user-images.githubusercontent.com/11477974/129084582-ca9135a5-37b7-4578-a94f-39a8d9f95702.png)


layer norm and softmax dont have opinfos yet but they are extremely straightforward for shape analysis

ZolotukhinM commented on torch/testing/_internal/common_methods_invocations.py:
```diff
@@ -6620,6 +6618,7 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
            dtypesIfCPU=floating_types_and(torch.bfloat16),
            dtypesIfCUDA=floating_types_and(torch.half, torch.bfloat16),
            sample_inputs_func=sample_inputs_softmax_variant,
+           assert_jit_shape_analysis=True,
```
It seems that you've added more shape functions than tests. Is it intentional?

eellison replied:
yea, see my other comment about softmax & layernorm being straightforward unary ops. happy to expand opinfo coverage for more complicate operators but would prefer not to do it for unary ops.

ZolotukhinM reviewed (approved):
Looks good!

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30347513).

Commit:
Update on "Add batch of unary functions"


Differential Revision: [D30347513](https://our.internmc.facebook.com/intern/diff/D30347513)

[ghstack-poisoned]

Commit:
Update on "Add batch of unary functions"


Differential Revision: [D30347513](https://our.internmc.facebook.com/intern/diff/D30347513)

[ghstack-poisoned]

Commit:
Update on "Add batch of unary functions"


Differential Revision: [D30347513](https://our.internmc.facebook.com/intern/diff/D30347513)

[ghstack-poisoned]

Commit:
Update on "Add batch of unary functions"


Differential Revision: [D30347513](https://our.internmc.facebook.com/intern/diff/D30347513)

[ghstack-poisoned]

Commit:
Update on "Add batch of unary functions"


Differential Revision: [D30347513](https://our.internmc.facebook.com/intern/diff/D30347513)

[ghstack-poisoned]

Commit:
Update on "Add batch of unary functions"


Differential Revision: [D30347513](https://our.internmc.facebook.com/intern/diff/D30347513)

[ghstack-poisoned]

Commit:
Update on "Add batch of unary functions"


Differential Revision: [D30347513](https://our.internmc.facebook.com/intern/diff/D30347513)

[ghstack-poisoned]

Commit:
Update on "Add batch of unary functions"


Differential Revision: [D30347513](https://our.internmc.facebook.com/intern/diff/D30347513)

[ghstack-poisoned]

Commit:
Update on "Add batch of unary functions"


Differential Revision: [D30347513](https://our.internmc.facebook.com/intern/diff/D30347513)

[ghstack-poisoned]

Commit:
Update on "Add batch of unary functions"


Differential Revision: [D30347513](https://our.internmc.facebook.com/intern/diff/D30347513)

[ghstack-poisoned]

Commit:
Update on "Add batch of unary functions"


Differential Revision: [D30347513](https://our.internmc.facebook.com/intern/diff/D30347513)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30347513).

Commit:
Update on "Add batch of unary functions"


Differential Revision: [D30347513](https://our.internmc.facebook.com/intern/diff/D30347513)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30347513).

Commit:
Update on "Add batch of unary functions"


Differential Revision: [D30347513](https://our.internmc.facebook.com/intern/diff/D30347513)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30347513).
